presto-hive/src/main/java/com/facebook/presto/hive/HiveTableLayoutHandle.java

```diff
@@ -147,7 +147,21 @@ protected HiveTableLayoutHandle(
         this.layoutString = requireNonNull(layoutString, "layoutString is null");
         this.requestedColumns = requireNonNull(requestedColumns, "requestedColumns is null");
         this.partialAggregationsPushedDown = partialAggregationsPushedDown;
-        this.appendRowNumberEnabled = appendRowNumberEnabled;
+        // if $row_id is requested, we need the row numbers whether they're explicitly requested or not
+        // TODO is it possible someone needs the row_numbers and row IDs, and thus we need a block for each?
```

Comment:
i think we need a plan for appendRowNumberEnabled and rowids. The append rownumber logic always seemed a bit hacky to me (we add a boolean to the table handle in our internal page source provider when we create the page source. but it feels like it was just put there as a handy way to pass things through and not because it necessarily is the best way to do it).  IMO the rowid design where you add a proper column is much nicer and the rowid is more useful.   I think we should work on having it replace the rownumber thing.

Reply:
Makes sense. Can you file an issue for that? I don't want to muck with the existing code more than I have to here.